data-r2dbc/src/test/groovy/io/micronaut/data/r2dbc/mysql/MySqlRepositorySpec.groovy

```diff
@@ -169,4 +170,51 @@ class MySqlRepositorySpec extends AbstractRepositorySpec implements MySqlTestPro
         return true
     }
 
+    void "test data with datetime fields"() {
+        given:
+        def defaultTimeZone = TimeZone.getDefault()
```

Comment:
@jchrys Just fyi that this test is failing with 1.0.4 version. Working ok with 1.0.3
With default timezone on Github CI (UTC) is working ok. Seems like resulted by this revert PR https://github.com/asyncer-io/r2dbc-mysql/pull/152/

Reply:
Thanks a lot for pinging me.
I actually intended for this change to occur because version 1.0.3 introduced a breaking change without a minor version update, which led to compatibility issue: [link](https://github.com/asyncer-io/r2dbc-mysql/issues/151). Therefore, I made the decision to revert it. However, we do plan to reintroduce the change with a minor version update very soon.

Reply:
Great, thanks!

Reply:
@radovanradic Thanks for your patience and reaching out about the date issue. 
I had initially planned to include the fix in version 1.1 but faced some delays. Due to comeback of an original  maintainer, I wanted to consult with him about the changes he introduced, which contributed to this issue, before proceeding with the modifications. Thank you for bringing this matter to our attention again. Although it's an estimate, I anticipate that the fix will be implemented and released sometime in March.

Reply:
Hello @radovanradic,

I ran this test case and placed breakpoints in `r2dbc-mysql`. Then I noticed that date times are encoded with `LocalDateTime`, and decoded with `ZonedDateTime` by `r2dbc-mysql` default type mappings.

Therefore, if they are persisted, they lose their time zone and use the database time zone on subsequent retrievals.

Currently, `r2dbc-mysql` will always query the time zone of the current MySQL session if the configuration option `serverZoneId` is not set. On the contrary, `mysql-connector-j` will use JVM local time zone by default, it will ignore the real time zone of session.

`r2dbc-mysql` would persist date times with time zone conversion, and retrieve date times with session time zone. It is equivalent to `preserveInstant=true&connectionTimeZone=SERVER&forceConnectionTimeZone=false` in `mysql-connector-j`. For comparison, the default value of `mysql-connector-j` is `preserveInstant=true&connectionTimeZone=LOCAL&forceConnectionTimeZone=false`.

I'm not sure I really understand it, but I prefer to persist date times with their original time zone, and retrieve date times with their "real" time zone (the session time zone).

BTW, I'm now trying to align the timezone behavior between `r2dbc-mysql` and `mysql-connector-j`, is there anything I can help with?

Reply:
Basically we used dev-miku r2dbc mysql library and when switched to asyncer.io r2dbc mysql it was expected that tests will be compatible and pass. But like talked above, there were issues with tests with timestamp fields, it was fixed in 1.0.3 but then reverted (as it was breaking compatibility). @jchrys told that next minor version might have the fix we had in 1.0.3

Reply:
Yep, this is also part of the time zone behavior alignment. See also asyncer-io/r2dbc-mysql#166 .

> For example, if we are using JPA (3.2.0) + JDBC (8.2.0), create an entity with a `ZonedDateTime` property and try to `findAll`, we will see `com.mysql.cj.jdbc.result.ResultSetImpl.getTimestamp(int, Calendar)` (line 975) is called, it is not `ResultSet.getObject(int)` or `ResultSet.getObject(int, Class<T>)`. Which means, JPA will always get `java.sql.Timestamp` first, then convert it to `ZonedDateTime` after that. Actually, `mysql-connector-j` returns `LocalDateTime` as default type of `DATETIME` in `ResultSetImpl.getObject(int)`.

Actually, `mysql-connector-j` is using `LocalDateTime` for `DATETIME` and `ZonedDateTime` for `TIMESTAMP` by default.

So this problem may be a more complicated problem, we may have difficulty solving the time zone problem using `LocalDateTime`. Because it does not contain any time zone.

The problem is easily solved if we persist the date time using `ZonedDateTime`. WDYT?

Reply:
I won't say anything new, just thought since previous library was stable enough and no reported bugs in this area (I assume) then why not do the same here? We didn't have issues with it and we use the same tests for other jdbc/r2dbc libs.